ambry-tools/src/main/java/com.github.ambry/store/BlobIdTransformer.java

```diff
@@ -0,0 +1,153 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.messageformat.BlobData;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.messageformat.MessageFormatErrorCodes;
+import com.github.ambry.messageformat.MessageFormatException;
+import com.github.ambry.messageformat.MessageFormatRecord;
+import com.github.ambry.messageformat.PutMessageFormatInputStream;
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.github.ambry.messageformat.MessageFormatRecord.*;
+
+
+/**
+ * Transformer implementation that replaces BlobIds in messages with
+ * the converted value from the StoreKeyConverter
+ */
+public class BlobIdTransformer implements StoreCopier.Transformer {
+
+  private final StoreKeyConverter storeKeyConverter;
+  private final StoreKeyFactory storeKeyFactory;
+
+  public BlobIdTransformer(StoreKeyConverter storeKeyConverter, StoreKeyFactory storeKeyFactory) {
+    this.storeKeyFactory = Objects.requireNonNull(storeKeyFactory, "storeKeyFactory must not be null");
+    this.storeKeyConverter = Objects.requireNonNull(storeKeyConverter, "storeKeyConverter must not be null");
+  }
+
+  /**
+   * Takes the input message and transforms it by possibly
+   * replacing the store key and account/container IDs
+   * with a new store key and account/container IDs
+   * @param message
+   * @return
+   * @throws Exception
+   */
+  @Override
+  public StoreCopier.Message transform(StoreCopier.Message message) throws Exception {
+    Objects.requireNonNull(message, "message must not be null");
+    Objects.requireNonNull(message.getMessageInfo(), "message's messageInfo must not be null");
+    Objects.requireNonNull(message.getStream(), "message's inputStream must not be null");
+    StoreKey oldStoreKey = message.getMessageInfo().getStoreKey();
+    StoreKey newStoreKey;
+    List<StoreKey> list = Collections.singletonList(oldStoreKey);
+    Map<StoreKey, StoreKey> storeKeyToStoreKey = storeKeyConverter.convert(list);
+    newStoreKey = storeKeyToStoreKey.get(oldStoreKey);
+    if (newStoreKey == null) {
+      return null;
+    }
+    return newMessage(message.getStream(), newStoreKey);
+  }
+
+  /**
+   * Extracts the message header from the input stream
+   * @param stream input stream from which the message header will be read
+   * @return message header from stream, will be of either Message_Header_Version_V1 or Message_Header_Version_V2
+   * @throws IOException
+   * @throws MessageFormatException
+   */
+  private MessageFormatRecord.MessageHeader_Format getMessageHeader(InputStream stream)
```

Comment:
Is this method necessary? Even if it is, I think this would belong in `MessageFormat`.

Reply:
I can take a look at this as part of the other PR.